_studio/mfx_lib/decode/vc1/include/mfx_vc1_decode.h

```diff
@@ -222,7 +222,7 @@ class MFXVideoDECODEVC1 : public VideoDECODE, public MfxCriticalErrorHandler
     mfxU32                           m_BufOffset;
 
     // to synchronize async and sync parts
-    UMC::Mutex                       m_guard;
+    std::recursive_mutex             m_guard;
```

Comment:
```recursive_mutex``` is an antipatters which often indicates bad architecture design

Reply:
Agree about bad design, but I wouldn't introduce redesign to this PR

Reply:
@akokoshn let's then split this PR to two, to have all the stuff related to ```recursive_mutex``` separetely